lib/internal/stream_base_commons.js

```diff
@@ -81,8 +81,40 @@ function afterWriteDispatched(self, req, err, cb) {
   }
 }
 
+function _onread(handle, stream, nread, buf) {
+  stream[kUpdateTimer]();
+
+  if (nread > 0 && !stream.destroyed) {
```

Comment:
This is not the same as the previous http2 implementation, which executed the logic within this block when `nread >= 0`, not just `nread > 0`.

Reply:
I don’t think omitting 0-sized reads is an issue.